.github/workflows/pullRequestWorkflow.yml

```diff
@@ -0,0 +1,22 @@
+name: Pull Request Workflow
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: CPU Details
+        run: lscpu
+      - uses: actions/checkout@v4
+      - uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: "1.1.34"
+      - run: bun install --frozen-lockfile
+      - run: bun run check:environment
+      # - run: bun run check:package-version
+      #   env:
+      #     GITHUB_API_URL: ${{ env.GITHUB_API_URL }}
+      #     GITHUB_REF_NAME: ${{ env.GITHUB_REF_NAME }}
+      #     GITHUB_REPOSITORY: ${{ env.GITHUB_REPOSITORY }}
+      #     GITHUB_TOKEN: ${{ github.token }}
```

Comment:
This is a convenience check during CI that will fail if the PR doesn't increment the `package.json` version and match the latest version in `CHANGELOG.md`. So instead of needing to remember to do it manually and having a reminder in the pull request template, it's now automated. However, since this is the first version being published, it doesn't work so needs to be disabled until the 2nd PR.

Reply:
I'll make a follow-up PR to turn this on

Reply:
Are we sure we want this?

Forcing package version update for all merges may sound nice initially, but then it means no change can be merged without a release, which isn't ideal as there are valid changes that don't require releases, such as documentation only changes, local development only changes (local scripts, dev dependencies update).

It also prevents the ability of batching updates (though for this I think, given we have a low flow of contributions, it's also ok to use release branches)

Reply:
I find it very useful. Documentation or local development changes can be a patch release. Plenty of open source packages follow this philosophy. I see more upsides than downsides mainly because it makes the publishing process more foolproof and less error prone; in other words, you don't have to tell people what to do, it does it for you.

Reply:
Yeah I understand the benefits, I am personally not a big fan of libraries having releases for changes that don't affect end users, but definitely more an opinion than anything so all good.